roles/nvidia-dgx-firmware/tasks/update-firmware.yml

```diff
@@ -2,15 +2,15 @@
 
 - include: check-firmware.yml
 
-- name: Check if force udpate requested
+- name: Check if force update requested
   set_fact:
     force_parameter: '-f'
-  when: "force_update == 'true'"
+  when: force_update
 
 - name: Update firmware if required
   block:
     - name: Update firmware if required
-      command: "docker run --rm --privileged -v /:/hostfs -t -e auto=1 -e COLUMNS=200 -e LINES=300 {{ firmware_update_container }} update_fw {{ force_parameter }} {{ target_fw }}"
+      command: "docker run --rm -ti --privileged -v /:/hostfs -e auto=1 -e COLUMNS=200 -e LINES=300 {{ firmware_update_repo }}:{{ firmware_update_tag }} set_flags AUTO=1 update_fw {{ force_parameter }} {{ target_fw }}"
```

Comment:
Does this work on the latest firmware? I recall that some of these flags changed at one point.

Reply:
Yes, tested `BMC` update and `all`. Also tested running an update and forced update on already updated nodes. Test subjects were all DGX-1s at a variety of different fw versions.